File: swift-app/Sources/SufrixUI/Components/Money.swift
// Money formatting — minor units → a display string. Kept tiny and identical to
// the Kotlin `Money` so totals read the same on both platforms. (A locale-aware
// formatter can replace this later; the catalog wire is already integer minor.)
import Foundation

enum Money {
    /// "EGP 12.50" — `minor` units rendered with two decimals and the code.
    static func format(_ minor: Int64, _ code: String) -> String {
        let major = Double(minor) / 100
        let amount = String(format: "%.2f", major)
        let c = code.uppercased()
        return c.isEmpty ? amount : "\(c) \(amount)"
    }
}

File: kotlin-app/composeApp/src/commonMain/kotlin/app/sufrix/ui/Money.kt
package app.sufrix.ui

import kotlin.math.absoluteValue
import kotlin.math.roundToLong

// Money formatting — minor units → a display string. Identical to the Swift
// `Money` so totals read the same on both platforms.
object Money {
    /** "EGP 12.50" — [minor] units rendered with two decimals and the code. */
    fun format(minor: Long, code: String): String {
        val neg = minor < 0
        val cents = minor.absoluteValue
        val whole = cents / 100
        val frac = (cents % 100).toString().padStart(2, '0')
        val amount = "${if (neg) "-" else ""}$whole.$frac"
        val c = code.uppercase()
        return if (c.isEmpty()) amount else "$c $amount"
    }
}

// Round-trip helper kept here so both the amount field and money display agree.
internal fun Double.toMinorUnits(): Long = (this * 100).roundToLong()
